File: RemoveDuplicatesFromSortedArray.java

public class RemoveDuplicatesFromSortedArray {
	public static int removeDuplicates(int[] nums) {
		if (nums.length < 2)
	        return nums.length;
	    int cur = 0;
	    for (int i = 1; i < nums.length; i++) {
	        if (nums[i] != nums[cur]) {
	            nums[++cur] = nums[i];
	        }   
	    }   
	    return cur + 1;  
    }
	public static void main(String[] args) {
		int[] nums = {1, 1, 2};
		int n = removeDuplicates(nums);
		for (int i = 0; i < n; i++) {
			System.out.print(nums[i] + " ");
		}
		System.out.println();
	}
}

File: py_solution/remove_duplicates_from_sorted_array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) < 2:
            return len(nums)
        top = 0
        for i in range(1, len(nums)):
            if nums[top] != nums[i]:
                top += 1
                nums[top] = nums[i]
        return top + 1


if __name__ == "__main__":
    solution = Solution()
    nums = [1,1,2]
    print(solution.removeDuplicates(nums), nums)
